tests/test_variables.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# -*- coding: utf-8 -*-
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 NYU.
+#
+# ultraviolet is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+# Tests  to check configuration settings
+from invenio_app.factory import create_app
+
+
+# Check that configuration variables read correctly from os
+def test_var_assigned(monkeypatch):
+    """Mocking setting configuration using environment varuables"""
+    monkeypatch.setenv("APP_ALLOWED_HOSTS", "ultraviolet.dlib.nyu.edu")
+    monkeypatch.setenv(
+        "SQLALCHEMY_DATABASE_URI", "postgresql+psycopg2://test:test@somehost.com/test"
+    )
+    monkeypatch.setenv("SITE_UI_URL", "ultraviolet.dlib.nyu.edu")
+    monkeypatch.setenv("SITE_API_URL", "ultraviolet.dlib.nyu.edu/api")
+    app = create_app()
+    assert app.config.get("APP_ALLOWED_HOSTS") == "ultraviolet.dlib.nyu.edu"
+    assert (
+        app.config.get("SQLALCHEMY_DATABASE_URI")
+        == "postgresql+psycopg2://test:test@somehost.com/test"
+    )
+    assert app.config.get("SITE_UI_URL") == "ultraviolet.dlib.nyu.edu"
+    assert app.config.get("SITE_API_URL") == "ultraviolet.dlib.nyu.edu/api"
+
+
+# Check that default values are assign to configuration variables when not passed through os
+def test_var_noassigned():
+    """Mocking using default configuration"""
+    app = create_app()
+    assert app.config.get("APP_ALLOWED_HOSTS") == ["0.0.0.0", "localhost", "127.0.0.1"]
+    assert (
+        app.config.get("SQLALCHEMY_DATABASE_URI")
+        == "postgresql+psycopg2://nyu-data-repository:nyu-data-repository@localhost/nyu-data-repository"
+    )
+    assert app.config.get("RDM_RECORDS_DOI_DATACITE_ENABLED") == False
```

Comment:
Minor:

In Python, when comparing against **language constants** (`None`, `True`, `False`...), it is better to use `is`, instead of `==`. The reason for this is that `is` compares memory addresses, while `==` compares values. Comparing memory addresses is quicker.

Reply:
Added changes